simplq/src/components/pages/Admin/AdminPage.jsx

```diff
@@ -6,18 +6,44 @@ import { useDispatch, useSelector } from 'react-redux';
 import { selectQueueName } from 'store/selectedQueue';
 import Ribbon from 'components/common/Ribbon';
 import Tour from 'components/common/Tour';
-import { useGetSelectedQueue } from 'store/asyncActions';
+import { useGetSelectedQueue, useGetQueueInfo } from 'store/asyncActions';
+import { selectQueueInfo } from 'store/queueInfo';
 import HeaderSection from './AdminHeaderSection';
 import TokenList from './TokenList';
 import styles from './admin.module.scss';
 import SidePanel from './AdminSidePanel';
 import getToursteps from './getTourSteps';
+import PageNotFound from '../PageNotFound';
 
 const TIMEOUT = 10000;
 let timeoutId;
 
-export default (props) => {
+const isQueueDeleteStatusUnknown = (queueInfo) => !queueInfo || Object.keys(queueInfo).length === 0;
+const isQueueDeleted = (queueInfo) => queueInfo.status === 'DELETED';
+
+const AdminPage = (props) => {
   const queueId = props.match.params.queueId;
+  const queueInfo = useSelector(selectQueueInfo);
+  const dispatch = useDispatch();
+  const getQueueInfo = useCallback(useGetQueueInfo(), []);
+
+  useEffect(() => {
+    dispatch(getQueueInfo({ queueId }));
+  }, [dispatch, queueId, getQueueInfo]);
+
+  if (isQueueDeleteStatusUnknown(queueInfo)) {
+    return <></>;
+  }
+
+  if (isQueueDeleted(queueInfo)) {
+    return <PageNotFound />;
+  }
```

Comment:
in this case, let's use [setErrorPopupMessage](https://github.com/SimplQ/simplQ-frontend/blob/master/simplq/src/store/appSlice.js#L17) to show user a good message and then take the user to the home page using `useHistory`.

It might be better UX than showing the 404 page. What do you think?

cc @maaverik

Reply:
True. Explicitly saying that the queue is deleted is more meaningful. For this, just a popup would be as good as a new page.

Reply:
Sure. Will change this.

Reply:
Done